odpf/compass/v1beta1/service.proto

```diff
@@ -0,0 +1,436 @@
+
+  rpc CreateComment(CreateCommentRequest) returns (CreateCommentResponse) {
+    option (google.api.http) = {
+      post: "/v1beta1/discussions/{discussion_id}/comments"
+      body: "*"
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Create a comment of a discussion";
+      tags: ["Discussion","Comment"];
+    };
+  }
+
+  rpc GetAllComments(GetAllCommentsRequest) returns (GetAllCommentsResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/discussions/{discussion_id}/comments"
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Get all comments of a discussion";
+      tags: ["Discussion","Comment"];
+    };
+  }
+
+  rpc GetComment(GetCommentRequest) returns (GetCommentResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/discussions/{discussion_id}/comments/{id}"
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Get a comment of a discussion";
+      tags: ["Discussion","Comment"];
+    };
+  }
+
+  rpc UpdateComment(UpdateCommentRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      put: "/v1beta1/discussions/{discussion_id}/comments/{id}"
+      body: "*"
+    };
+    
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Update a comment of a discussion";
+      tags: ["Discussion","Comment"];
+    };
+  }
+
+  rpc DeleteComment(DeleteCommentRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1beta1/discussions/{discussion_id}/comments/{id}"
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Delete a comment of a discussion";
+      tags: ["Discussion","Comment"];
+    };
+  };
+};
+
+message GetAllDiscussionsRequest {
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+    json_schema: {
+      title: "GetAllDiscussionsRequest"
+      description: "Some parameters to fetch discussions"
+    }
+  };
+
+  string type = 1 [(validate.rules).string = { in: [ "openended", "issues", "qanda", "all" ], ignore_empty: true }];
+  string state = 2 [(validate.rules).string = { in: [ "open", "closed", "all" ], ignore_empty: true }];
+  string owner = 3 [(validate.rules).string.ignore_empty = true];
+  string assignee = 4 [(validate.rules).string.ignore_empty = true];
+  string asset = 5 [(validate.rules).string.ignore_empty = true];
+  string labels = 6 [(validate.rules).string.ignore_empty = true];
+  string sort = 7 [(validate.rules).string = { in: [ "created_at", "updated_at" ], ignore_empty: true }];
+  string direction = 8 [(validate.rules).string = { in: [ "asc", "desc" ], ignore_empty: true  }];
+  uint32 size = 9 [(validate.rules).uint32 = { gte: 0, ignore_empty: true } ];
+  uint32 offset = 10 [(validate.rules).uint32 = { gte: 0, ignore_empty: true }];
```

Comment:
how's this validation rules gonna be used?

Reply:
@rahmatrhd you can refer to this one https://github.com/envoyproxy/protoc-gen-validate#protoc-gen-validate-pgv

Reply:
Like this one in columbus 	https://github.com/odpf/columbus/blob/f1b47eb31b512b492fad28b845c237b422a84fd3/api/v1beta1/discussion.go#L26